src/v/cluster/cluster_discovery.cc

```diff
@@ -131,8 +136,39 @@ cluster_discovery::dispatch_node_uuid_registration_to_seeds(
     co_return ss::stop_iteration::no;
 }
 
-bool cluster_discovery::is_cluster_founder() const {
-    return config::node().seed_servers().empty();
+/*static*/ std::optional<node_id>
+cluster_discovery::get_cluster_founder_node_id() {
+    if (config::node().empty_seed_starts_cluster()) {
+        if (config::node().seed_servers().empty())
+            return node_id{0};
+        else
+            return {};
+    } else {
+        if (auto idx = get_node_index_in_seed_servers(); idx)
+            return node_id{*idx};
+        else
+            return {};
+    }
+}
+
+/*static*/ std::optional<int32_t>
+cluster_discovery::get_node_index_in_seed_servers() {
```

Comment:
should we include an information that we are looking for self id in the seed server list, not any node idx ?

Reply:
The info is here: https://github.com/dlex/redpanda/blob/333_cluster-bootstrap/src/v/cluster/cluster_discovery.cc#L73-L77

It will also log in case if `empty_seed_starts_cluster=true` and node_id is 0 for the root.